scripts/stac/imagery/tests/create_stac_test.py

```diff
@@ -479,3 +493,84 @@ def test_create_item_when_resupplying_with_changed_asset_file(subtests: SubTests
 
     with subtests.test(msg="assets.visual.updated"):
         assert item.stac["assets"]["visual"]["updated"] == current_datetime
+
+
+def test_get_items_to_replace() -> None:
+    published_items = [
+        {
+            "type": "Feature",
+            "id": "item_a",
+        },
+        {
+            "type": "Feature",
+            "id": "item_b",
+        },
+        {
+            "type": "Feature",
+            "id": "item_c",
+        },
+    ]
+    supplied_item = [
+        {
+            "type": "Feature",
+            "id": "item_d",
+        },
+        {
+            "type": "Feature",
+            "id": "item_b",
+        },
+    ]
+
+    items_to_replace = get_items_to_replace(published_items, supplied_item)
+    assert items_to_replace == [
+        {
+            "type": "Feature",
+            "id": "item_b",
+        }
+    ]
+
+
+def test_prepare_resupply(fake_collection_metadata: CollectionMetadata, fake_linz_slug: str, subtests: SubTests) -> None:
+    published_items = [
+        {"type": "Feature", "id": "item_a"},
+        {"type": "Feature", "id": "item_b"},
+        {"type": "Feature", "id": "item_c"},
+    ]
+    supplied_items: list[dict[str, Any]] = [
+        {"type": "Feature", "id": "item_d"},
+        {
+            "type": "Feature",
+            "id": "item_b",
+            "properties": {"start_datetime": "2024-09-02T12:00:00Z", "end_datetime": "2024-09-02T12:00:00Z"},
+        },
+    ]
+    links = [
+        {"rel": "item", "href": "./item_a.json"},
+        {"rel": "item", "href": "./item_b.json"},
+        {"rel": "item", "href": "./item_c.json"},
+        {"rel": "self", "href": "./collection.json"},
+    ]
+
+    collection = ImageryCollection(
+        fake_collection_metadata, any_epoch_datetime_string(), any_epoch_datetime_string(), fake_linz_slug
+    )
+    collection.stac["links"] = links
+    merged_items = prepare_resupply(collection, published_items, supplied_items)
+
+    with subtests.test("merged items"):
+        assert merged_items == [
+            {"type": "Feature", "id": "item_d"},
+            {
+                "type": "Feature",
+                "id": "item_b",
+                "properties": {"start_datetime": "2024-09-02T12:00:00Z", "end_datetime": "2024-09-02T12:00:00Z"},
+            },
+            {"type": "Feature", "id": "item_a"},
+            {"type": "Feature", "id": "item_c"},
+        ]
+
+    with subtests.test("links"):
+        assert collection.stac["links"] == [{"rel": "self", "href": "./collection.json"}]
+
+    with subtests.test("extent"):
```

Comment:
Note to myself, needs to actuall add an extent to the collection before running `prepare_resupply`